src/screens/CameraScreen.tsx

```diff
@@ -1,110 +1,97 @@
 import React, { useRef, useState } from 'react';
-import { View, StyleSheet, SafeAreaView, TouchableOpacity, Image, Alert } from 'react-native';
+import { View, StyleSheet, Image } from 'react-native';
 import { RNCamera } from 'react-native-camera';
-import { PermissionsAndroid, Platform } from 'react-native';
-import CameraRoll from '@react-native-community/cameraroll';
-import Icon from 'react-native-vector-icons/Ionicons';
+import { NativeStackNavigationProp } from '@react-navigation/native-stack';
+import { RouteProp } from '@react-navigation/native';
+import { useDispatch } from 'react-redux';
 
-const CameraScreen: React.FC = () => {
+import { CAMERA_SCREEN, MAIN_SCREEN, RootStackParamList } from '../../App';
+import { addPhoto } from '../../store/actions';
+import { CameraScreenButton } from '../components/CameraScreenButton';
+
+type CameraScreenProps = {
+  navigation: NativeStackNavigationProp<RootStackParamList, typeof CAMERA_SCREEN>;
+  route?: RouteProp<RootStackParamList, typeof CAMERA_SCREEN>;
+};
+
+const CameraScreen: React.FC<CameraScreenProps> = ({ navigation, route }) => {
   const cameraRef = useRef<RNCamera | null>(null);
-  const [lastPhotoUri, setLastPhotoUri] = useState<string | null>(null);
+  const [photoUri, setPhotoUri] = useState<string | undefined>(route?.params?.uri);
   const [flashOn, setFlashOn] = useState<boolean>(false);
+  const dispatch = useDispatch();
+
+  const isHereFromGallery = route?.params?.hereFromGallery;
+
+  const addPhotoHandler = () => {
```

Comment:
For more consistency with other callbacks, I suggest changing the name to just `savePhoto` (without handler)